client/lib/browser-storage/index.ts

```diff
@@ -146,6 +186,34 @@ export async function getStoredItem< T >( key: string ): Promise< T | undefined
 	return await idbGet( key );
 }
 
+/**
+ * Get all stored items.
+ *
+ * @param pattern The pattern to match on returned item keys.
+ * @returns A promise with the stored key/value pairs as an object. Empty if none.
+ */
+export async function getAllStoredItems( pattern?: RegExp ): Promise< StoredItems > {
+	if ( shouldBypass ) {
+		return await bypassGetAll( pattern );
+	}
+
+	const idbSupported = await supportsIDB();
+	if ( ! idbSupported ) {
+		const results: StoredItems = {};
+		for ( let i = 0; i < window.localStorage.length; i++ ) {
```

Comment:
I suspect that nowadays `Object.entries( window.localStorage )` works well, too, and should even work on IE11 with the CoreJS polyfills.

Reply:
Yes, it does seem to work! Thanks!